src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -1071,38 +1071,45 @@ static AnalysisCacheData()
             }
 
             string cacheFileName = "ModuleAnalysisCache";
-            if (ExperimentalFeature.EnabledExperimentalFeatureNames.Count > 0)
+            // When multiple copies of pwsh are on the system, they should use their own copy of the cache.
+            // Append hash of `$PSHOME` to cacheFileName.
+            byte[] hashBytes;
+            using (var sha1 = SHA1.Create())
             {
-                // If any experimental features are enabled, we cannot use the default cache file because those
-                // features may expose commands that are not available in a regular powershell session, and we
-                // should not cache those commands in the default cache file because that will result in wrong
-                // auto-completion suggestions when the default cache file is used in another powershell session.
-                //
-                // Here we will generate a cache file name that represent the combination of enabled feature names.
-                // We first convert enabled feature names to lower case, then we sort the feature names, and then
-                // compute an SHA1 hash from the sorted feature names. We will use a short SHA name (first 8 chars)
-                // to generate the cache file name.
-                int index = 0;
-                string[] featureNames = new string[ExperimentalFeature.EnabledExperimentalFeatureNames.Count];
-                foreach (string featureName in ExperimentalFeature.EnabledExperimentalFeatureNames)
-                {
-                    featureNames[index++] = featureName.ToLowerInvariant();
-                }
+                hashBytes = sha1.ComputeHash(Encoding.UTF8.GetBytes(Utils.DefaultPowerShellAppBase));
```

Comment:
Why use a cryptographic hash at all?  Why not use `.GetHashCode()`?

Reply:
We need a stable result but the `GetHashCode` could be randomized.

Another question why now we don't take in account `ExperimentalFeature` in the file name?

Reply:
I believe `GetHashCode()` in .NET Core is now randomized as @iSazonov stated and we need something stable

Reply:
https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=crc32&type=

but I don't understand how much we will win.

Reply:
I presume that this has to be calculated at most every startup.  Switching to a less intensive hash algorithm will reduce our startup time which is a big factor in how people perceive performance.

cc @daxian-dbw

Reply:
@TravisEz13 and I discussed this offline yesterday and I agree that we should go with CRC. The main gain is to potentially avoid loading `System.Security.Cryptography.Algorithms.dll` at powershell startup time, plus there is also gain in CPU time when running the hash algorithm. I will do this change as part of my startup improvement work.

Reply:
Got this chart from StackOverflow about the collision rate of CRC32:
![image](https://user-images.githubusercontent.com/127450/48641696-778b1980-e98f-11e8-8888-821891590c1a.png)

In our scenario, hashing would be done on each powershell core installation on a machine, and in case of any experimental features is enabled, the combination of feature names. I think 100 different hash inputs should be a fair large number for a local machine environment in our scenario, and thus the collision rate would be less than 1 in a million, which should be good.

Reply:
We could use SSE from System.Runtime.Intrinsics.Experimental. We get this in runtime after .Net Core 3.0 release.